evm/chains/base.py

```diff
@@ -480,6 +488,26 @@ def from_genesis_header(cls, chaindb, genesis_header):
     #
     # Mining and Execution API
     #
+    def apply_transaction(self, transaction):
+        """
+        Applies the transaction to the current tip block.
+        """
+        vm = self.get_vm()
+        block, receipt, computation = vm.apply_transaction(transaction)
+        receipts = block.get_receipts(self.chaindb) + [receipt]
```

Comment:
This feels a little weird. We're getting back the block after the transaction, but it doesn't have the receipts that were also produced as a part of the transaction and some of the final header fields.

I think I get where you're eventually going: that we shouldn't return a block at all, but the building pieces that are needed to form the block. It is just hard to track which of those changes are still going in the block and which aren't. Maybe that's fine for this PR, where you're just trying to enable this one feature of incremental building, and come back for a reorg later.

A `TransactionDiff` is continuing to look like a nice encapsulation, which would include all the things that change in the block, plus the receipt, and the computation.

Reply:
Agreed, though I'm going to push fixing this out to a later step.